### PR TITLE
Nested Topics

### PR DESCRIPTION
**Pull request by EvilVir**
Type: **feature**

## Summary
Added ability to create nested hierarchy of topics/cms pages.
In topic's edit page you can now set ParentTopicId value, if that value is set then page will be nested under given parent in top menu.

## Testing
1. Create some Topic in CMS, check **include in top menu**.
2. Create another Topic in CMS but this time in **Parent topic** field input the **SystemName** of the topic created in step 1. Also check **include in top menu**. Repeat couple of times.
3. Refresh the client page, you will see inactive-link to the topic from step 1 and when you hover over it dropdown will show up with link from step2.
4. Switch to mobile view, link from step 1 should be visible and once clicked menu should switch to view of links from step 2 with additional back button at the top.
5. Click back button, you should be returned to previous menu.

### CODE REVIEW

File: Grand.Web/web.config [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <aspNetCore requestTimeout="23:00:00" processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" forwardWindowsAuthToken="false" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" startupTimeLimit="3600">
       <environmentVariables>
         <environmentVariable name="ASPNETCORE_ENVIRONMENT" value="Development" />
+        <environmentVariable name="COMPLUS_ForceENC" value="1" />
```
Comment (PatrykPorabik) [inline Grand.Web/web.config]: Hi,

Please remove the web.config file from the pull request.

Comment (EvilVir) [reply Grand.Web/web.config]: Should me removed now.

**Comment by PatrykPorabik** [comment]
By the way, maybe it will be better to enter the system name of parent topic, instead of IDs? I can't image that store owner will search for the ID in the store. It's better to use the system name, imho.

**Comment by EvilVir** [comment]
But it is SystemName used (see step 2. of testing instructions ;)

**Comment by PatrykPorabik** [comment]
Hi Kuba,

Finally, I've implemented this feature on my local environment, because we want to implement it in the incoming 4.50 version of GrandNode. I was testing the nested topic feature on mobile and I've noticed that when you want to back to main topic, jQuery throws error in the dev console. Also when we open the child topic on mobile, the opening canvas is on the 70% width of the whole screen, is it a bug or a feature? :) [Here is the error](https://i.imgur.com/mvRB4mb.jpg)

**Comment by EvilVir** [comment]
Hm that's new :) I'm sure I didn't touched any JavaScript to implement that thing. I'll check that this week and let you know. TBH I mostly used this feature in own theme, that dropped all this jsoffcanvas thingy in favor of pure Bootstrap 4 navbar, so maybe I missed some test cases.

**Comment by PatrykPorabik** [comment]
Kuba,

Did you have chance to check it?

**Comment by KrzysztofPajak** [comment]
Conflicting files

**Pull request closed without merging**